cmd/gardenlet/app/app.go

```diff
@@ -460,6 +468,38 @@ func (g *garden) cleanupOrphanedExtensionsServiceAccounts(ctx context.Context, g
 	return flow.Parallel(taskFns...)(ctx)
 }
 
+// TODO(Kostov6): Remove this code after v1.91 has been released.
+func (g *garden) cleanupGRMSecretFinalizers(ctx context.Context, seedClient client.Client) error {
+	mrs := &resourcesv1alpha1.ManagedResourceList{}
+	if err := seedClient.List(ctx, mrs); err != nil {
+		if meta.IsNoMatchError(err) {
+			// resources.gardener.cloud/v1alpha1 CRD not yet applied
```

Comment:
You can add a log entry with proper explanation instead of comment that does not explain much (why the CRD is not yet applied for example).

```suggestion
			log.Info("Received a 'no match error' while trying to list managed resources. Will assume that the managed resources CRD is not yet installed (for example new Seed creation) and will skip cleaning up GRM finalizers.")

```